packages/xrpc-server/src/server.ts

```diff
@@ -481,30 +484,27 @@ export class Server {
   }
 }
 
-function isHandlerSuccess(v: HandlerOutput): v is HandlerSuccess {
-  return handlerSuccess.safeParse(v).success
-}
-
-function isHandlerPipeThrough(v: HandlerOutput): v is HandlerPipeThrough {
-  if (v === null || typeof v !== 'object') {
-    return false
-  }
-  if (!isString(v['encoding']) || !(v['buffer'] instanceof ArrayBuffer)) {
-    return false
-  }
-  if (v['headers'] !== undefined) {
-    if (v['headers'] === null || typeof v['headers'] !== 'object') {
-      return false
-    }
-    if (!Object.values(v['headers']).every(isString)) {
-      return false
+const createValidator =
+  <T>(schema: ZodSchema<T>) =>
+  (v: unknown): v is T =>
+    schema.safeParse(v).success
```

Comment:
we have a helper function for this that we use in most places in `common-web/check`

`check.is(v, schema)`

Reply:
Added a `check.create(schema)` as the semantics I needed are a bit different